presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package me.androidbox.presentation.agenda.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import me.androidbox.domain.authentication.preference.PreferenceRepository
+import me.androidbox.presentation.agenda.screen.AgendaScreenEvent
+import me.androidbox.presentation.agenda.screen.AgendaScreenState
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    private val preferenceRepository: PreferenceRepository
+) : ViewModel() {
+
+    private val _agendaScreenState = MutableStateFlow(AgendaScreenState())
+    val agendaScreenState = _agendaScreenState.asStateFlow()
+
+    init {
+        getAuthenticatedUser()
+    }
+
+    fun onAgendaScreenEvent(agendaScreenEvent: AgendaScreenEvent) {
+        when(agendaScreenEvent) {
+            is AgendaScreenEvent.OnDateChanged -> {
+                _agendaScreenState.update { agendaScreenState ->
+                    agendaScreenState.copy(
+                        displayMonth = agendaScreenEvent.date.month.toString()
+                    )
+                }
+            }
+        }
+    }
+
+    private fun getAuthenticatedUser() {
+        viewModelScope.launch {
+            preferenceRepository.retrieveCurrentUserOrNull()?.let { authenticatedUser ->
+                _agendaScreenState.update { agendaScreenState ->
+                    agendaScreenState.copy(
+                        usersInitials = authenticatedUser.fullName.extractInitials()
+                    )
+                }
+            }
+        }
+    }
```

Comment:
@philipplackner @MKiperszmid After the user enters the Agenda screen fetching the user from shared preferences to show their initials in the top bar.

Reply:
Yea that's correct